### PR TITLE
Blog styling

### PR DESCRIPTION
**Pull request by maelle**
Design.

### CODE REVIEW

**Comment by maelle** [comment]
@gaborcsardi you can review the post (as opposed to the design mess) when you have time.

**Comment by maelle** [comment]
post = https://deploy-preview-30--admiring-allen-9a00b2.netlify.com/2019/11/21/readmes/

**Comment by maelle** [comment]
Reg Rmd pieces to be used in vignettes, it seems one shouldn't store them in vignettes/, here's my workaround https://twitter.com/ma_salmon/status/1199979670026014720?s=20 I'll update the post if the hack is ok.

File: content/post/2019-11-21-readmes.Rmarkdown [hunk 1]
```diff
@@ -0,0 +1,416 @@
...
+
+* in the GitHub repo description, 
+
+* in your introduction at a social event (ok, maybe not).
+
+Other parts of a pitch are good talk fodder, blog post introductions, vignette sections, etc. Therefore, the time you spend pitching your package in the best possible way is a gift that'll keep on giving, to your users and you.  
+
+## What is a good README
+
+In the Art of README, noffle [includes a checklist](https://github.com/noffle/art-of-readme#bonus-the-readme-checklist); and rOpenSci dev guide [features guidance about the README](https://devguide.ropensci.org/building.html#readme). Now, what about good READMEs in the wild? In this section, we'll have a look at a small sample of READMEs.
```
Comment (gaborcsardi) [inline content/post/2019-11-21-readmes.Rmarkdown]: and _the_ rOpenSci dev guide

File: content/post/2019-11-21-readmes.Rmarkdown [hunk 2]
```diff
@@ -0,0 +1,416 @@
...
+                URL) %>%
+  dplyr::mutate(path = urltools::path(URL)) %>%
+  dplyr::mutate(path = stringr::str_remove(path, "\\/$")) %>%
+  tidyr::separate(path, sep = "\\/", into = c("owner", "repo"))
+str(meta)
+```
+
+At this point we have `r nrow(meta)` packages with `r length(unique(meta$URL))` unique GitHub repo URLs, pfiew.
+
+We'll then extract their [preferred README from GitHub V3 API](https://developer.github.com/v3/repos/contents/#get-the-readme). Some of them won't even have one so we'll lose them from the sample.
```
Comment (gaborcsardi) [inline content/post/2019-11-21-readmes.Rmarkdown]: from _the_ GitHub V3 API

File: content/post/2019-11-21-readmes.Rmarkdown [hunk 3]
```diff
@@ -0,0 +1,416 @@
...
+  readme <- try(gh("GET /repos/:owner/:repo/readme",
+                   owner = owner, repo = repo),
+                silent = TRUE)
+  if(inherits(readme, "try-error")){
+    return(NULL)
+  }
+  
+  lines <- suppressWarnings(
+    readLines(readme$download_url
+              )
```
Comment (gaborcsardi) [inline content/post/2019-11-21-readmes.Rmarkdown]: code indentation is weird here

File: content/post/2019-11-21-readmes.Rmarkdown [hunk 4]
```diff
@@ -0,0 +1,416 @@
...
+library("ggplot2")
+
+ggplot(readmes) +
+  geom_dotplot(aes(lines_no), binwidth = 5) +
+  xlab("No. of lines") +
+  scale_y_continuous(NULL, breaks = NULL) +
+  hrbrthemes::theme_ipsum(base_size = 16,
+                          axis_title_size = 16) +
+  ggtitle(glue::glue("Number of lines in a sample of {nrow(readmes)} READMEs"))
+```
```
Comment (gaborcsardi) [inline content/post/2019-11-21-readmes.Rmarkdown]: Is there something wrong with the y scale here? Is it better to bin them, and show a simple histogram instead?

File: content/post/2019-11-21-readmes.Rmarkdown [hunk 5]
```diff
@@ -0,0 +1,416 @@
...
+    length
+}
+
+loc <- readmes %>%
+  dplyr::group_by(repo, owner) %>%
+  dplyr::summarise(loc = get_code_lines(xml_readme[[1]]))
+
+```
+
+The number of lines of code of READMEs range from `r min(loc$loc)` (for `r sum(loc$loc == min(loc$loc))` READMEs) to `r max(loc$loc)` with a median of `r median(loc$loc)`. The README with the most lines of code is `r glue::glue("https://github.com/{loc$owner[loc$loc == max(loc$loc)]}/{loc$repo[loc$loc == max(loc$loc)]}")`.
```
Comment (gaborcsardi) [inline content/post/2019-11-21-readmes.Rmarkdown]: You can append `#readme` to link to the actual README.

File: content/post/2019-11-21-readmes.Rmarkdown [hunk 6]
```diff
@@ -0,0 +1,416 @@
...
+
+```
+</details>
+
+In practice,
+
+```{r r42}
+print_readme_structure(structure$structure[42][[1]])
+print_readme_structure(structure$structure[7][[1]])
+print_readme_structure(structure$structure[1][[1]])
```
Comment (gaborcsardi) [inline content/post/2019-11-21-readmes.Rmarkdown]: I think there is a bug in the tree drawing:

```
## A Teradata Backend for dplyr
##          ├─── Koji Makiyama (@hoxo-m)
##    ├─ 1 Overview
##    ├─ 2 Installation
##    ├─ 3 Details
##       ├── 3.1 Usage
##       ├── 3.2 Translatable functions
##          ├─── 3.2.1 lubridate friendly functions
##          ├─── 3.2.2 Treat Boolean
##          ├─── 3.2.3 to_timestamp()
##          └─── 3.2.4 cut()
##       └── 3.3 Other useful functions
##          └─── 3.3.1 blob_to_string()
##    └─ 4 Related work
```

File: content/post/2019-11-21-readmes.Rmarkdown [hunk 7]
```diff
@@ -0,0 +1,416 @@
...
+
+Since `roxygen2`'s latest release, you can include md/Rmd in manual pages. What a handy thing for, say, your [package-level documentation](https://usethis.r-lib.org/reference/use_package_doc.html)! [Refer to `roxygen2`'s documentation](https://roxygen2.r-lib.org/articles/rd.html#including-external--rmd-md-files).
+
+In the R files use
+
+```r
+#' @includeRmd man/rmdhunks/child.Rmd
+```
+
+#### Table of content
```
Comment (gaborcsardi) [inline content/post/2019-11-21-readmes.Rmarkdown]: Table of _contents_?

File: content/post/2019-11-21-readmes.Rmarkdown [hunk 8]
```diff
@@ -0,0 +1,416 @@
...
+
+If your README is a bit long you might need to add a table of contents at the top, [as done for `cli` README](https://github.com/r-lib/cli/blob/ad6410aee086f374e974eb7e794fa0d475815320/README.Rmd#L6). In a pkgdown website, the README does not get a table of content in the sidebar, which might be an argument for keeping it small as opposed to articles that do get a table of contents in the sidebar.
```
Comment (gaborcsardi) [inline content/post/2019-11-21-readmes.Rmarkdown]: Well, the cli README does not have one any more. :(

File: content/post/2019-11-21-readmes.Rmarkdown (hunk 4 above)
Comment (maelle) [reply content/post/2019-11-21-readmes.Rmarkdown]: I don't understand?

File: content/post/2019-11-21-readmes.Rmarkdown (hunk 6 above)
Comment (maelle) [reply content/post/2019-11-21-readmes.Rmarkdown]: what would you expect here?

File: content/post/2019-11-21-readmes.Rmarkdown (hunk 8 above)
Comment (maelle) [reply content/post/2019-11-21-readmes.Rmarkdown]: ok I'll use pkgsearch.

File: content/post/2019-11-21-readmes.Rmarkdown (hunk 5 above)
Comment (maelle) [reply content/post/2019-11-21-readmes.Rmarkdown]: awesome idea

File: content/post/2019-11-21-readmes.Rmarkdown (hunk 4 above)
Comment (gaborcsardi) [reply content/post/2019-11-21-readmes.Rmarkdown]: nolinesplot-1

Much of this plot is empty. I was wondering if a histogram would show more information.

File: content/post/2019-11-21-readmes.Rmarkdown (hunk 6 above)
Comment (gaborcsardi) [reply content/post/2019-11-21-readmes.Rmarkdown]: The broken line on the left to continue down to "Related work".

**Pull request merged**